### ISSUE TITLE
why use whole dataset as one batch?

### ISSUE THREAD

**Issue by ShawnX4869** [czifan/DeepSurv.pytorch#9]
Hello,bro.
At line 39&41 in 'main.py', it looks like trainset/testset takes whole file as one batch cause of 'batch_size=train_dataset.__len__()'. But why? The mask size is increased with sample length right? Can i seperate it into more batches, like 512 per batch?
Looking forward ur reply, it would be very helpful for me.
Thanks.

**Comment by wuchengyu123** [comment]
I encounter the same question. In fact, if I tune the train batch_size in to a small number, for example 32, it will always go 0.5 at concordance index and I have no idea why.

**Comment by CQTsang** [comment]
hey bro. 
I turn the train batch_size into 32 and it gets a better c_index when using METABRIC dataset, epoch=5. I suppose it's a coincidence when using smaller batch, the mask in NegativeLogLikelihood turns into most 0, giving a bad impact on the calculation of the loss.

**Comment by czifan** [comment]
I agree with CQTsang's option. Thanks for CQTsang!

Because the NegativeLogLikelihood loss is calculated within a batch, a too-small batch size would cause some samples without compared objects (also indicated as "the mask in NegativeLogLikelihood turns into most 0" by CQTsang). Training all data into one batch in this code because the used dataset is small, a larger and more suitable batch size, like 32 or 64, could be considered in your experiments.
